### PR TITLE
Remove split_queries_by_interval and forward_headers_list

### PR DESCRIPTION
**Pull request by shantanualsi**
…range config
Removed split_queries_by_interval and forward_headers_list from query_range config

The fields were deprecated and hence going to be removed in the next major version

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #<issue number>

**Special notes for your reviewer**:

**Checklist**
- [x] Reviewed the [`CONTRIBUTING.md`](https://github.com/grafana/loki/blob/main/CONTRIBUTING.md) guide (**required**)
- [ ] Documentation added
- [ ] Tests updated
- [x] `CHANGELOG.md` updated
  - [ ] If the change is worth mentioning in the release notes, add `add-to-release-notes` label
- [x] Changes that require user attention or interaction to upgrade are documented in `docs/sources/setup/upgrade/_index.md`
- [ ] For Helm chart changes bump the Helm chart version in `production/helm/loki/Chart.yaml` and update `production/helm/loki/CHANGELOG.md` and `production/helm/loki/README.md`. [Example PR](https://github.com/grafana/loki/commit/d10549e3ece02120974929894ee333d07755d213)

### CODE REVIEW

File: docs/sources/setup/upgrade/_index.md [hunk 1]
```diff
@@ -102,6 +102,8 @@ You can use `--keep-empty` flag to retain them.
 3. `s3.sse-encryption` is removed. AWS now defaults encryption of all buckets to SSE-S3. Use `sse.type` to set SSE type. 
 4. `ruler.wal-cleaer.period` is removed. Use `ruler.wal-cleaner.period` instead.
 5. `experimental.ruler.enable-api` is removed. Use `ruler.enable-api` instead.
+6. `query_range.split_queries_by_interval` is now removed. Use the flag in `limits_config` instead.
+7. `query_range.forward_headers_list` is now removed
```
Comment (ashwanthgoli) [inline docs/sources/setup/upgrade/_index.md:106]:
```suggestion
7. `frontend.forward-headers-list` CLI flag and its corresponding YAML setting are removed.
```

File: docs/sources/setup/upgrade/_index.md [hunk 2]
```diff
@@ -102,6 +102,8 @@ You can use `--keep-empty` flag to retain them.
 3. `s3.sse-encryption` is removed. AWS now defaults encryption of all buckets to SSE-S3. Use `sse.type` to set SSE type. 
 4. `ruler.wal-cleaer.period` is removed. Use `ruler.wal-cleaner.period` instead.
 5. `experimental.ruler.enable-api` is removed. Use `ruler.enable-api` instead.
+6. `query_range.split_queries_by_interval` is now removed. Use the flag in `limits_config` instead.
```
Comment (ashwanthgoli) [inline docs/sources/setup/upgrade/_index.md:105]:
```suggestion
6. `split_queries_by_interval` is removed from `query_range` YAML section. You can instead configure it in [Limits Config](/docs/loki/latest/configuration/#limits_config).
```

**Review by ashwanthgoli — APPROVED** [review]
lgtm, left a couple of nits to be consistent with the previous comments in the upgrade notes

**Comment by shantanualsi** [comment]
thanks @ashwanthgoli , changed the docs as per suggestions :)

File: CHANGELOG.md [hunk 3]
```diff
@@ -67,6 +67,7 @@
 * [10073](https://github.com/grafana/loki/pull/10073) **sandeepsukhani,salvacorts,vlad-diachenko** Support attaching non-indexed labels to log lines.
 * [10378](https://github.com/grafana/loki/pull/10378) **shantanualsi** Remove deprecated `ruler.wal-cleaer.period`
 * [10380](https://github.com/grafana/loki/pull/10380) **shantanualsi** Remove `experimental.ruler.enable-api` in favour of `ruler.enable-api`
+* [10395](https://github.com/grafana/loki/pull/10395/) **shantanualshi** Remove deprecated split_queries_by_interval and forward_headers_list configs in query_range
```
Comment (chaudum) [inline CHANGELOG.md:70]: nit:

```suggestion
* [10395](https://github.com/grafana/loki/pull/10395/) **shantanualshi** Remove deprecated `split_queries_by_interval` and `forward_headers_list` configuration options in the `query_range` section
```

**Review by chaudum — APPROVED** [review]

**Pull request merged**